### PR TITLE
codegen: Add NDValidation, generation script

### PR DESCRIPTION
**Pull request by AlexDBlack**

### CODE REVIEW

File: codegen/src/main/stubs/org/nd4j/linalg/api/buffer/DataType.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package org.nd4j.linalg.api.buffer;
+
+public enum DataType {
```
Comment (treo) [inline codegen/src/main/stubs/org/nd4j/linalg/api/buffer/DataType.java:3]: can you stub this out?

I want to make sure that stubs only have the minimum amount of things in there,  so if we ever have to change things, we know exactly what we are actually directly referencing

File: codegen/generate.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+mvn clean package -DskipTests
+java -cp target/codegen-1.0.0-SNAPSHOT-shaded.jar org.nd4j.codegen.cli.CLI -dir ../../deeplearning4j -namespaces "$@"
```
Comment (treo) [inline codegen/generate.sh]: Can you also add a a sentence or two and an example under the usage section in the readme?

**Review by treo — CHANGES_REQUESTED** [review]
Mostly good, just two trivial things I'd like to be addressed :)

File: codegen/src/main/stubs/org/nd4j/linalg/api/buffer/DataType.java (hunk 1 above)
Comment (AlexDBlack) [reply codegen/src/main/stubs/org/nd4j/linalg/api/buffer/DataType.java:3]: Datatype enums values are required, at least for BOOL and UTF8
https://github.com/KonduitAI/dl4j-dev-tools/blob/626ec1e1348fbc64566b3bf6f9a46ee3148850a9/codegen/src/main/resources/java/utilClasses/NDValidation.java#L40

Same thing for isIntType and isFPType

One option is to simply stub NDValidation here, have the real one in ND4J, and never copy back... I'd be fine with that instead.

File: codegen/generate.sh (hunk 2 above)
Comment (AlexDBlack) [reply codegen/generate.sh]: Done

File: codegen/src/main/stubs/org/nd4j/linalg/api/buffer/DataType.java (hunk 1 above)
Comment (treo) [reply codegen/src/main/stubs/org/nd4j/linalg/api/buffer/DataType.java:3]: For the record:
We discussed this, and NDValidation is meant to be stubbed here, and not copied into ND4J. The copying mechanism was just a crutch while nd4j didn't have NDValidation.

**Pull request merged**